src/components/views/settings/ChangePassword.tsx

```diff
@@ -82,48 +82,64 @@ export default class ChangePassword extends React.Component<IProps, IState> {
         };
     }
 
-    private onChangePassword(oldPassword: string, newPassword: string): void {
+    private async onChangePassword(oldPassword: string, newPassword: string): Promise<void> {
         const cli = MatrixClientPeg.get();
 
-        if (!this.props.confirm) {
-            this.changePassword(cli, oldPassword, newPassword);
-            return;
-        }
+        const serverSupportsControlOfDevicesLogout: boolean = await cli.doesServerSupportLogoutDevices();
```

Comment:
```suggestion
        const serverSupportsControlOfDevicesLogout = await cli.doesServerSupportLogoutDevices();
```

We prefer implicit types where the right side of the assignment is well-typed